packages/agreement-process/test/deleteAgreement.test.ts

```diff
@@ -98,6 +103,107 @@ describe("delete agreement", () => {
     ).not.toContain(agreement.consumerDocuments[1].path);
   });
 
+  it("should succeed when requester is Consumer Delegate and the Agreement is in a deletable state", async () => {
+    vi.spyOn(fileManager, "delete");
+    const agreementId = generateId<AgreementId>();
+    const consumerDocuments = [
+      getMockConsumerDocument(agreementId, "doc1"),
+      getMockConsumerDocument(agreementId, "doc2"),
+    ];
+
+    const agreement = {
+      ...getMockAgreement(),
+      id: agreementId,
+      state: randomArrayItem(agreementDeletableStates),
+      consumerDocuments,
+    };
+
+    const authData = getRandomAuthData(agreement.consumerId);
+
+    const delegation = getMockDelegation({
+      kind: delegationKind.delegatedConsumer,
+      eserviceId: agreement.eserviceId,
+      delegatorId: agreement.consumerId,
+      delegateId: authData.organizationId,
+      state: delegationState.active,
+    });
+
+    await addOneAgreement(agreement);
+    await addOneDelegation(delegation);
```

Comment:
Consider also adding some random more delegations to make sure the right one is retrieved, you can reuse the method that sando added here: https://github.com/pagopa/interop-be-monorepo/pull/1275/files#diff-34c4465d023488a00d0496f065ad1a816ab3e56b780634fb324772309c93e5eaR179

(it's also duplicated in activateAgreement tests so maybe you could extract it in the agreement test utils).